rerun_py/src/dataframe.rs

```diff
@@ -0,0 +1,635 @@
+            store: &self.store,
+            cache: &self.cache,
+        }
+    }
+
+    /// Convert a `ViewContentsLike` into a `ViewContentsSelector`.
+    ///
+    /// ```python
+    /// ViewContentsLike = Union[str, Dict[str, Union[ComponentLike, Sequence[ComponentLike]]]]
+    /// ```
+    ///
+    /// We cant do this with the norma `FromPyObject` mechanisms because we want access to the
+    /// `QueryEngine` to resolve the entity paths.
+    fn extract_contents_expr(
+        &self,
+        expr: Bound<'_, PyAny>,
+    ) -> PyResult<re_chunk_store::ViewContentsSelector> {
+        let engine = self.engine();
+
+        if let Ok(expr) = expr.extract::<String>() {
+            // `str`
+
+            let path_filter = EntityPathFilter::parse_strict(&expr, &Default::default())
+                .map_err(|err| PyValueError::new_err(err.to_string()))?;
+
+            let contents = engine
+                .iter_entity_paths(&path_filter)
+                .map(|p| (p, None))
+                .collect();
+
+            Ok(contents)
+        } else if let Ok(dict) = expr.downcast::<PyDict>() {
+            // `Union[ComponentLike, Sequence[ComponentLike]]]`
+
+            let mut contents = ViewContentsSelector::default();
+            for (key, value) in dict {
+                let key = key.extract::<String>()?;
+
+                let path_filter = EntityPathFilter::parse_strict(&key, &Default::default())
+                    .map_err(|err| PyValueError::new_err(err.to_string()))?;
+
+                let components: BTreeSet<ComponentName> =
+                    if let Ok(component) = value.extract::<ComponentLike>() {
+                        std::iter::once(component.0).collect()
+                    } else if let Ok(components) = value.extract::<Vec<ComponentLike>>() {
+                        components.into_iter().map(|c| c.0).collect()
+                    } else {
+                        return Err(PyTypeError::new_err(
+                            "ViewContentsLike input must be a string or a list of strings.",
+                        ));
+                    };
+
+                contents.append(
+                    &mut engine
+                        .iter_entity_paths(&path_filter)
+                        .map(|p| (p, Some(components.clone())))
+                        .collect(),
+                );
+            }
+
+            Ok(contents)
+        } else {
+            return Err(PyTypeError::new_err("ViewContentsLike input must be..."));
+        }
+    }
+}
+
+#[pymethods]
+impl PyRecording {
+    fn schema(&self) -> PySchema {
+        PySchema {
+            schema: self.store.schema(),
+        }
+    }
+
+    #[pyo3(signature = (
+        *,
+        timeline,
+        contents
+    ))]
+    fn view(
+        slf: Bound<'_, Self>,
+        timeline: &str,
+        contents: Bound<'_, PyAny>,
+    ) -> PyResult<PyRecordingView> {
+        let borrowed_self = slf.borrow();
+
+        // Look up the type of the timelin
```

Comment:
Staying ahead of `typos`...
```suggestion
        // Look up the type of the timeline